### DISCUSSION TITLE
re-saving the image using this library changes a signature done by another sdk.

### DISCUSSION THREAD

**Discussion by amrali21** [dlemstra/Magick.NET#1338] [Help] [answered]
I am re-saving my images using this library to get better quality.

this is the code I use: 

```
  string imgFile = @"\\D\TempImages\original.exr";
  string testDirectory = @"\\D\TempImages\";

  byte[] byteArr = File.ReadAllBytes(imgFile);
  MagickImage mi = new MagickImage(byteArr, MagickFormat.Exr);

  File.WriteAllBytes(testDirectory + "new.exr", mi.ToByteArray(MagickFormat.Exr));
```
"original.exr" in the code above is created by an image sdk (which is not open-source) that puts a signature on images. so i'm having a problem where sdk does not recognize the images anymore after I re-save them using the example above. also the size of the image changes from 1,020kb to 1,356kb.

so my question is can I re-save the images using your library without changing some feature in the image so it's signature doesn't get messed up? thank you.

**Comment by dlemstra** [comment, marked as answer]
I don't know what you mean by signature. Not sure how they are protecting their file but if it is a "good" signature they would use the file data to set the signature and then there is no real way to preserve the signature. And I am also not sure if the `C` api of the openexr library can be used to add a signature if they just used a string attribute with their company name. And the increment might be due to a different compression method, and maybe that is why you are unable to read the image? Maybe you should reach out to the vendor about this?

**Reply by amrali21** [reply]
Thank you for the answer. I will reach out to the vendor as well. Just to be sure, is there no way to reverse the process in the example above, and remove any changes made on the image so it's recognizable again by the SDK (no matter how they're doing the signature).

I'm asking because I re-saved many images this way, and deleted the original ones. So now all I have is the images that are not recognized by the sdk.

**Reply by dlemstra** [reply]
There is no way to reverse the process. All you can do is use your backups or recycle bin if you are lucky.

**Reply by amrali21** [reply]
Okay, thank you for help ;)
